android/beagle/src/main/java/br/com/zup/beagle/android/data/serializer/adapter/ActionAnalyticsConfigAdapterFactory.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.data.serializer.adapter
+
+import br.com.zup.beagle.newanalytics.ActionAnalyticsConfig
+import br.com.zup.beagle.newanalytics.ActionAnalyticsProperties
+import br.com.zup.beagle.android.data.serializer.BeagleMoshi.moshi
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonReader
+import com.squareup.moshi.JsonWriter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import java.lang.reflect.Type
+
+class ActionAnalyticsConfigAdapterFactory : JsonAdapter.Factory {
```

Comment:
Again, if you change the name of the action, this class could be changed too.
```suggestion
class AnalyticsActionConfigAdapterFactory : JsonAdapter.Factory {
```

Reply:
Nice suggestion, I'll change the name!